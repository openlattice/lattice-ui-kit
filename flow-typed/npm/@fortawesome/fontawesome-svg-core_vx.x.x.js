// flow-typed signature: 39887e5e6c6c32861704232d9719cd14
// flow-typed version: <<STUB>>/@fortawesome/fontawesome-svg-core_v1.2.4/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/fontawesome-svg-core'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/fontawesome-svg-core' {
  declare module.exports: any;
}
