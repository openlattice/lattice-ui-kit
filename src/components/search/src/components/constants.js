import { List, Map, fromJS } from 'immutable';

const mockFilterFields = [
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'reportType',
    label: 'Report type',
    options: ['Crisis Template', 'Follow-up'],
  },
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'badges',
    label: 'Badges',
    options: ['Officer Safety', 'Substance use'],
  },
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'submitter',
    label: 'Submitter',
    options: [
      'squidward@bubblebowl.com',
      'smitty@werbenjagermanjensen.com',
      'wumbology@thestudyofwumbo.com',
      'enigma@innermachinations.com'
    ],
  },
];

const mockSearchResultsForReports = List([
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety'],
    submitter: 'squidward@bubblebowl.com'
  }),
  Map({
    reportType: 'Follow-up',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'squidward@bubblebowl.com',
  }),
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'smitty@werbenjagermanjensen.com'
  }),
  Map({
    reportType: 'Follow-up',
    badges: ['Substance use'],
    submitter: 'wumbology@thestudyofwumbo.com'
  }),
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'enigma@innermachinations.com'
  }),
  Map({
    reportType: 'Follow-up',
    submitter: 'enigma@innermachinations.com'
  }),
]);

const mockSearchResultsForPeople = List([
  Map({
    lastName: 'Werbenjagermanjensen',
    firstName: 'Smitty',
    middleName: '',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Fish',
    dob: '02/22/2002',
    identifier: '365ba027-0c71-47d1-a5f4-61346ed96d8c',
  }),
  Map({
    lastName: 'Thornton',
    firstName: 'Cameron',
    middleName: 'Orville',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Black / African American',
    dob: '01/05/1982',
    identifier: '565ba027-0c71-47d1-a5f4-61346ed96d8c',
  }),
  Map({
    lastName: 'Gerner',
    firstName: 'Timothy',
    middleName: 'Pernell',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Native Hawaiian or Other Pacific Islander',
    dob: '10/02/1991',
    identifier: '465ba027-0c71-47d1-a5f4-61346ed96d8c',
  })
]);

export {
  mockFilterFields,
  mockSearchResultsForReports,
  mockSearchResultsForPeople
};
