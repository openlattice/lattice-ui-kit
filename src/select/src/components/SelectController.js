// @flow
import { Component } from 'react';
import type { Node } from 'react';

import isFunction from 'lodash/isFunction';

import DropdownIndicator from './styled/DropdownIndicator';
import SelectOption from './SelectOption';
import ValueContainer from './styled/ValueContainer';

import type { ReactSelectEvent, ReactSelectOption, ReactSelectValue } from '../../types';

type Props = {
  dropdownIcon ? :Node;
  hideDropdownIcon ? :boolean;
  inputIcon ? :Node;
  isMulti :boolean;
  onChange :(value :any, event :ReactSelectEvent) => void;
  options :ReactSelectOption[] | any[];
  render :(onChange :Function, rawValues :any) => Node;
  useRawValues :boolean;
  value :any;
}

class SelectController extends Component<Props> {

  static defaultProps = {
    dropdownIcon: undefined,
    hideDropdownIcon: false,
    inputIcon: undefined,
  };

  handleChangeRawValues = (selectedOption :ReactSelectValue, event :ReactSelectEvent) => {
    const { onChange } = this.props;
    const rawValue = this.getValue(selectedOption);
    if (isFunction(onChange)) {
      onChange(rawValue, event);
    }
  }

  getValue = (selectedOption :ReactSelectValue) :any => {
    const { isMulti } = this.props;
    if (isMulti && Array.isArray(selectedOption)) {
      return selectedOption.map((option) => option.value);
    }
    if (selectedOption && !Array.isArray(selectedOption)) {
      return selectedOption.value;
    }
    return undefined;
  }

  getOptionsFromRawValue = () :ReactSelectOption[] | ReactSelectOption | void => {
    const { isMulti, options, value } = this.props;
    const optionsByValue = options.reduce((acc, option) => {
      acc[option.value] = option;
      return acc;
    }, {});
    if (isMulti && Array.isArray(value)) {
      return value.map((v) => this.getOption(v, optionsByValue));
    }
    if (value !== undefined && value !== null) {
      return this.getOption(value, optionsByValue);
    }

    return undefined;
  }

  getOption = (value :any, optionsByValue :Object) :ReactSelectOption => {
    if (optionsByValue[value]) {
      return optionsByValue[value];
    }
    return {
      label: `${value}`,
      value
    };
  }

  composeProps = (props :Object) :Object => {

    const defaultComponents :Object = {
      Option: SelectOption,
      ValueContainer,
      DropdownIndicator
    };

    const components = { ...defaultComponents, ...props.components };

    return { ...props, components };
  }

  renderWithDataOverrides = () => {
    const {
      onChange,
      render,
      useRawValues,
      value,
      ...rest
    } = this.props;

    const overrideProps :Object = {
      ...rest, onChange: this.handleChangeRawValues
    };

    if (value !== undefined && value !== null) {
      overrideProps.value = this.getOptionsFromRawValue();
    }

    const composedProps = this.composeProps(overrideProps);
    return render(composedProps);
  }

  render() {
    const { useRawValues, render, ...rest } = this.props;
    const composedProps = this.composeProps({ ...rest });
    return (
      <>
        {
          useRawValues
            ? this.renderWithDataOverrides()
            : render(composedProps)
        }
      </>
    );
  }
}

export default SelectController;
