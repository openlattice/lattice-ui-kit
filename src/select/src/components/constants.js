export const LATTICE_SELECT = 'lattice-select';
export const OPTIONS = [
  { label: '0', value: 'zero' },
  { label: '1', value: 'one' },
  { label: '2', value: 'two' },
  { label: '3', value: 'three' },
  { label: '4', value: 'four' },
];
