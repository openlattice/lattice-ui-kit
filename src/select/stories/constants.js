// @flow

/* eslint-disable max-len */
const REACT_SELECT_USAGE :string = 'Select and Creatable components are simple wrappers that pass OpenLattice styling to their respective react-select components. All props provided to these wrappers are passed through allowing full use of react-select functionality.';

const stateOptions = [
  { value: 'AL', label: 'Alabama' },
  { value: 'AK', label: 'Alaska' },
  { value: 'AS', label: 'American Samoa' },
  { value: 'AZ', label: 'Arizona' },
  { value: 'AR', label: 'Arkansas' },
  { value: 'CA', label: 'California' },
  { value: 'CO', label: 'Colorado' },
  { value: 'CT', label: 'Connecticut' },
  { value: 'DE', label: 'Delaware' },
  { value: 'DC', label: 'District Of Columbia' },
  { value: 'FM', label: 'Federated States Of Micronesia' },
  { value: 'FL', label: 'Florida' },
  { value: 'GA', label: 'Georgia' },
  { value: 'GU', label: 'Guam' },
  { value: 'HI', label: 'Hawaii' },
  { value: 'ID', label: 'Idaho' },
  { value: 'IL', label: 'Illinois' },
  { value: 'IN', label: 'Indiana' },
  { value: 'IA', label: 'Iowa' },
  { value: 'KS', label: 'Kansas' },
  { value: 'KY', label: 'Kentucky' },
  { value: 'LA', label: 'Louisiana' },
  { value: 'ME', label: 'Maine' },
  { value: 'MH', label: 'Marshall Islands' },
  { value: 'MD', label: 'Maryland' },
  { value: 'MA', label: 'Massachusetts' },
  { value: 'MI', label: 'Michigan' },
  { value: 'MN', label: 'Minnesota' },
  { value: 'MS', label: 'Mississippi' },
  { value: 'MO', label: 'Missouri' },
  { value: 'MT', label: 'Montana' },
  { value: 'NE', label: 'Nebraska' },
  { value: 'NV', label: 'Nevada' },
  { value: 'NH', label: 'New Hampshire' },
  { value: 'NJ', label: 'New Jersey' },
  { value: 'NM', label: 'New Mexico' },
  { value: 'NY', label: 'New York' },
  { value: 'NC', label: 'North Carolina' },
  { value: 'ND', label: 'North Dakota' },
  { value: 'MP', label: 'Northern Mariana Islands' },
  { value: 'OH', label: 'Ohio' },
  { value: 'OK', label: 'Oklahoma' },
  { value: 'OR', label: 'Oregon' },
  { value: 'PW', label: 'Palau' },
  { value: 'PA', label: 'Pennsylvania' },
  { value: 'PR', label: 'Puerto Rico' },
  { value: 'RI', label: 'Rhode Island' },
  { value: 'SC', label: 'South Carolina' },
  { value: 'SD', label: 'South Dakota' },
  { value: 'TN', label: 'Tennessee' },
  { value: 'TX', label: 'Texas' },
  { value: 'UT', label: 'Utah' },
  { value: 'VT', label: 'Vermont' },
  { value: 'VI', label: 'Virgin Islands' },
  { value: 'VA', label: 'Virginia' },
  { value: 'WA', label: 'Washington' },
  { value: 'WV', label: 'West Virginia' },
  { value: 'WI', label: 'Wisconsin' },
  { value: 'WY', label: 'Wyoming' },
];

const paragraph = `
October arrived, spreading a damp chill over the grounds and into the castle. Madam Pomfrey, the nurse, was kept busy by a sudden spate of colds among the staff and students. Her Pepperup potion worked instantly, though it left the drinker smoking at the ears for several hours afterward. Ginny Weasley, who had been looking pale, was bullied into taking some by Percy. The steam pouring from under her vivid hair gave the impression that her whole head was on fire. Raindrops the size of bullets thundered on the castle windows for days on end; the lake rose, the flower beds turned into muddy streams, and Hagrid's pumpkins swelled to the size of garden sheds. Oliver Wood's enthusiasm for regular training sessions, however, was not dampened, which was why Harry was to be found, late one stormy Saturday afternoon a few days before Halloween, returning to Gryffindor Tower, drenched to the skin and splattered with mud.
`;

export {
  REACT_SELECT_USAGE,
  paragraph,
  stateOptions,
};
