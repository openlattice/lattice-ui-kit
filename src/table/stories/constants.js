const TABLE_HEADERS = [
  { key: 'name', label: 'Subject Name', cellStyle: { width: '50%' } },
  { key: 'dob', label: 'Date of Birth' },
  { key: 'manager', label: 'Manager' },
  { key: 'lastUpdated', label: 'Last Updated' },
  { key: 'id', label: 'ID', sortable: false },
];

const TABLE_DATA = [
  {
    name: 'Eugene Krabs',
    dob: '1942-11-30',
    manager: 'Spongebob',
    lastUpdated: '2019-09-01',
    id: '1'
  },
  {
    name: 'Squidward Tentacles',
    dob: '',
    manager: 'Smitty',
    lastUpdated: '2019-08-29',
    id: '2'
  },
  {
    name: 'Sandy Cheeks',
    dob: '1987-11-17',
    manager: 'Patrick',
    lastUpdated: '2019-07-30',
    id: '3'
  },
  {
    name: 'Larry the Lobster',
    dob: '1975-06-23',
    manager: 'Patchy',
    lastUpdated: '2019-08-16',
    id: '4'
  },
  {
    name: 'Sheldon Plankton',
    dob: '1942-11-30',
    manager: 'Spongebob',
    lastUpdated: '2019-08-20',
    id: '5'
  },
  {
    name: 'Mrs. Puff',
    dob: '',
    manager: 'Spongebob',
    lastUpdated: '2019-08-21',
    id: '6'
  },
  {
    name: 'Flying Dutchman',
    dob: '1678-04-20',
    manager: 'Spongebob',
    lastUpdated: '2019-06-21',
    id: '7'
  },
];

export {
  TABLE_HEADERS,
  TABLE_DATA,
};
