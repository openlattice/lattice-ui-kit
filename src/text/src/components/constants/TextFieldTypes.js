/*
 * @flow
 */

type TextFieldTypeEnum = {|
  TEXT :'text';
  TEXT_AREA :'textarea';
|};
type TextFieldType = $Values<TextFieldTypeEnum>;

// TODO: look into using Immutable.Map() or other possible "enum" libraries
const TextFieldTypes :TextFieldTypeEnum = Object.freeze({
  TEXT: 'text',
  TEXT_AREA: 'textarea',
});

export default TextFieldTypes;
export type {
  TextFieldType,
  TextFieldTypeEnum,
};
