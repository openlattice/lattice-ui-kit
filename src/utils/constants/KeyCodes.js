/*
 * @flow
 */

type KeyCodeEnum = {|
  ENTER :'Enter';
  ESCAPE :'Escape';
|};
type KeyCode = $Values<KeyCodeEnum>;

// TODO: look into using Immutable.Map() or other possible "enum" libraries
const KeyCodes :KeyCodeEnum = Object.freeze({
  ENTER: 'Enter',
  ESCAPE: 'Escape',
});

export default KeyCodes;
export type {
  KeyCode,
  KeyCodeEnum,
};
