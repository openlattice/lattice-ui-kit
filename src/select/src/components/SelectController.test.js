import React from 'react';

import ReactSelect from 'react-select';
import toJson from 'enzyme-to-json';
import { faSearch } from '@fortawesome/pro-regular-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { mount, shallow } from 'enzyme';

import SelectController from './SelectController';
import { OPTIONS } from './constants';

// eslint-disable-next-line react/jsx-props-no-spreading
const mockComponent = ((selectProps) => <ReactSelect {...selectProps} />);

describe('Select', () => {

  describe('snapshot', () => {

    test('render matches snapshot', () => {
      const wrapper = shallow(<SelectController render={mockComponent} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

  });

  describe('getOptionsFromRawValue', () => {
    test('should return undefined if uncontrolled', () => {
      const wrapper = shallow(
        <SelectController
            options={OPTIONS}
            render={mockComponent}
            useRawValues />
      );

      const instance = wrapper.instance();
      const { getOptionsFromRawValue } = instance;

      const computedOptions = getOptionsFromRawValue();
      expect(computedOptions).toBeUndefined();
    });

    test('should return option from value', () => {
      const wrapper = shallow(
        <SelectController
            options={OPTIONS}
            render={mockComponent}
            useRawValues
            value="zero" />
      );

      const instance = wrapper.instance();
      const { getOptionsFromRawValue } = instance;

      const computedOptions = getOptionsFromRawValue();
      expect(computedOptions).toEqual(OPTIONS[0]);
    });

    test('should return options from values', () => {
      const wrapper = shallow(
        <SelectController
            isMulti
            options={OPTIONS}
            render={mockComponent}
            useRawValues
            value={['zero', 'one']} />
      );

      const instance = wrapper.instance();
      const { getOptionsFromRawValue } = instance;

      const computedOptions = getOptionsFromRawValue();
      expect(computedOptions).toEqual(OPTIONS.slice(0, 2));
    });
  });

  describe('useRawValues', () => {

    test('should not pass value if uncontrolled', () => {
      const wrapper = mount(
        <SelectController
            options={OPTIONS}
            render={mockComponent}
            useRawValues />
      );

      const reactSelectProps = wrapper.find(ReactSelect).props();
      expect(reactSelectProps).not.toHaveProperty('value');
    });

    test('useRawValues should transform raw values into options', () => {
      const wrapper = mount(
        <SelectController
            options={OPTIONS}
            render={mockComponent}
            useRawValues
            value="zero" />
      );

      const reactSelectValue = wrapper.find(ReactSelect).props().value;
      expect(reactSelectValue).toEqual(OPTIONS[0]);
    });

    test('isMulti > useRawValues should transform array of values into array of options', () => {
      const wrapper = mount(
        <SelectController
            isMulti
            options={OPTIONS}
            render={mockComponent}
            useRawValues
            value={['zero', 'one']} />
      );

      const reactSelectValue = wrapper.find(ReactSelect).props().value;
      expect(reactSelectValue).toEqual(OPTIONS.slice(0, 2));
    });

    test('do not pass useRawValues through render', () => {
      const wrapper = mount(
        <SelectController
            options={OPTIONS}
            render={mockComponent}
            useRawValues
            value="zero" />
      );
      const reactSelectProps = wrapper.find(ReactSelect).props();
      expect(reactSelectProps.useRawValues).toBeUndefined();
    });

    test('useRawValues should override onChange', () => {
      const wrapper = mount(<SelectController useRawValues render={mockComponent} />);
      const reactSelectProps = wrapper.find(ReactSelect).props();
      const { handleChangeRawValues } = wrapper.instance();
      expect(reactSelectProps.onChange).toEqual(handleChangeRawValues);
    });

  });

  describe('handleChangeRawValues', () => {
    test('handleChangeRawValues should return undefined for invalid onChange', () => {
      const wrapper = shallow(
        <SelectController
            useRawValues
            render={mockComponent}
            onChange={false} />
      );
      const instance = wrapper.instance();
      const { handleChangeRawValues } = instance;
      const mockEvent = { action: 'select-option' };
      const result = handleChangeRawValues(OPTIONS[0], mockEvent);

      expect(result).toEqual(undefined);
    });

    test('handleChangeRawValues should transform option to value', () => {
      const mockOnChange = jest.fn();
      const wrapper = shallow(
        <SelectController
            useRawValues
            render={mockComponent}
            onChange={mockOnChange} />
      );

      const instance = wrapper.instance();
      const getValueSpy = jest.spyOn(instance, 'getValue');
      instance.forceUpdate();

      const { handleChangeRawValues } = instance;
      const mockEvent = { action: 'select-option' };
      handleChangeRawValues(OPTIONS[0], mockEvent);

      expect(getValueSpy).toHaveBeenCalledWith(OPTIONS[0]);
      expect(mockOnChange).toHaveBeenCalledWith(OPTIONS[0].value, mockEvent);
    });

    test('handleChangeRawValues should transform options to values', () => {
      const mockOnChange = jest.fn();
      const wrapper = shallow(
        <SelectController
            isMulti
            onChange={mockOnChange}
            render={mockComponent}
            useRawValues />
      );

      const instance = wrapper.instance();
      const getValueSpy = jest.spyOn(instance, 'getValue');
      instance.forceUpdate();

      const { handleChangeRawValues } = instance;
      const mockEvent = { action: 'select-option' };

      const selectedOptions = OPTIONS.slice(0, 2);
      handleChangeRawValues(selectedOptions, mockEvent);

      const expectedValue = selectedOptions.map((option) => option.value);

      expect(getValueSpy).toHaveBeenCalledWith(selectedOptions);
      expect(mockOnChange).toHaveBeenCalledWith(expectedValue, mockEvent);
    });

    test('handleChangeRawValues should return undefined for empty selection', () => {
      const mockOnChange = jest.fn();
      const wrapper = shallow(
        <SelectController
            isMulti
            onChange={mockOnChange}
            render={mockComponent}
            useRawValues />
      );

      const instance = wrapper.instance();
      const getValueSpy = jest.spyOn(instance, 'getValue');
      instance.forceUpdate();

      const { handleChangeRawValues } = instance;
      const mockEvent = { action: 'clearValue' };

      const selectedOptions = undefined;
      handleChangeRawValues(selectedOptions, mockEvent);

      expect(getValueSpy).toHaveBeenCalledWith(selectedOptions);
      expect(mockOnChange).toHaveBeenCalledWith(undefined, mockEvent);
    });
  });

  describe('getOption', () => {
    test('getOption should return value from optionsByValue', () => {
      const wrapper = shallow(
        <SelectController
            render={mockComponent}
            useRawValues />
      );

      const instance = wrapper.instance();
      const { getOption } = instance;

      const newOption = getOption('zero', { zero: OPTIONS[0] });
      expect(newOption).toEqual(OPTIONS[0]);
    });

    test('getOption should return new option object if not found', () => {
      const wrapper = shallow(
        <SelectController
            render={mockComponent}
            useRawValues />
      );

      const instance = wrapper.instance();
      const { getOption } = instance;

      const newOption = getOption('DNE', {});
      expect(newOption).toEqual({ label: 'DNE', value: 'DNE' });
    });
  });

  describe('icon', () => {
    test('FontAwesome IconDefinition should render FontAwesomeIcon', () => {
      const wrapper = mount(
        <SelectController
            icon={faSearch}
            render={mockComponent} />
      );
      expect(wrapper.find(FontAwesomeIcon).props().icon).toEqual(faSearch);
    });

    test('should render custom icon', () => {
      const customIcon = () => <FontAwesomeIcon icon={faSearch} pulse />;
      const wrapper = mount(
        <SelectController
            icon={customIcon}
            render={mockComponent} />
      );
      expect(wrapper.find(FontAwesomeIcon).prop('icon')).toEqual(faSearch);
    });

    test('invalid icon renders default DropdownIndicator', () => {
      const wrapper = mount(
        <SelectController
            icon={false}
            render={mockComponent} />
      );
      expect(wrapper.find('DropdownIndicator').exists()).toBe(true);
    });

  });

});
