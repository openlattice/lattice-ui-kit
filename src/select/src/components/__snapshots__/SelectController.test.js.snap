// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select snapshot render matches snapshot 1`] = `
<Fragment>
  <StateManager
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
  />
</Fragment>
`;
